### ISSUE TITLE
Collectd::Plugin::Aggregation::Aggregator fails on 5.2

### ISSUE THREAD

**Issue by gytisgreitai** [voxpupuli/puppet-collectd#596]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 3.8.7
- Ruby: 1.9.3p484
- Distribution: Ubuntu 14.04
- Module version: 5.2

## How to reproduce (e.g Puppet code you use)
 ```
class { 'collectd::plugin::aggregation':
      aggregators => {
        cpu => {
          plugin           => 'cpu',
          type             => 'cpu',
          groupby          => ["Host", "TypeInstance"],
          calculateaverage => true,
        },
      }
    }
```
## What are you seeing
```
Error: Invalid parameter type on Collectd::Plugin::Aggregation::Aggregator[cpu] at xxx/client.pp:45 on node xx.xx.com
Error: Invalid parameter type on Collectd::Plugin::Aggregation::Aggregator[cpu] at xxx/client.pp:45 on node xx.xx.com

```
## What behaviour did you expect instead
It works fine on 5.1 and generates correct config

**Comment by juniorsysadmin** [comment]
@gytisgreitai Are you certain that you didn't do a typo and use `aggregator` rather than `aggregators` because that is what the error message seems to be indicating.

**Comment by gytisgreitai** [comment]
@juniorsysadmin yes. I have the exact same code as shown here, that works on 5.1, and fails on 5.2.
Since opening this issue I've upgraded to puppet 4.8.1 so the error has more details:

` Error: Evaluation Error: Error while evaluating a Resource Statement, Collectd::Plugin::Aggregation::Aggregator[cpu]: has no parameter named 'type'  at xxx/collectd/manifests/plugin/aggregation.pp:19 on node yyyyy`

And this leads us to this commit https://github.com/voxpupuli/puppet-collectd/commit/c0faa91ffdd7ffdd6d0bc815e001fddb340dd21e

So probably the README.md must be changed to use `agg_type` instead of type

Created Pull request for that https://github.com/voxpupuli/puppet-collectd/pull/606

**Comment by juniorsysadmin** [comment]
Docs fixed by #606
